### PR TITLE
depends: build libevent with CMake

### PR DESCRIPTION
**Pull request by fanquake**
Switches libevent in depends to be built with CMake.

### CODE REVIEW

**Comment by theuni** [comment]
I'm confused about this error. sigaddset should come from `<signal.h>`, which is included from `"evsignal-internal.h"`.

**Comment by luke-jr** [comment]
This doesn't seem like a good enough reason to use untested/unreleased code... :/

**Comment by theuni** [comment]
I believe (@fanquake can confirm) that the idea was to test the full merge, and if we liked the CMake build parts we'd patch them in and drop the rest.

**Comment by fanquake** [comment]
> and if we liked the CMake build parts we'd patch them in and drop the rest.

Yep. From the OP:
> Depending on the outcome, we could possibly pull a (minified) patch set into depends.

**Comment by fanquake** [comment]
Rebased. Dropped the final patch commit, as that change happened upstream. Also bumped to current master.

**Comment by hebasto** [comment]
Is something required to undraft this PR?

**Comment by fanquake** [comment]
We aren't getting much from master, so this is now just a switch to CMake. Once this lands, with CMake, a (minimal) depends + Core build will be possible entirely without autotools.

**Comment by fanquake** [comment]
> Is something required to undraft this PR?

Fixing the broken Windows build. Which is now done.

**Comment by fanquake** [comment]
Guix Build (aarch64):
```bash
db8e2595bb6b4d7088860f1df0c952be13a4ae3db3396a0c3a42de98165529bc  guix-build-a5180bffbe78/output/aarch64-linux-gnu/SHA256SUMS.part
e90d5520e9bec9efe81387012d6b9ddd189ed7f4737726d825cd8077c4c0b50b  guix-build-a5180bffbe78/output/aarch64-linux-gnu/bitcoin-a5180bffbe78-aarch64-linux-gnu-debug.tar.gz
a0b5ee10285feddbc964a694c134dcd0bb9193065985423063a353c2e518442b  guix-build-a5180bffbe78/output/aarch64-linux-gnu/bitcoin-a5180bffbe78-aarch64-linux-gnu.tar.gz
d49594130fd85f0c9016bab6aebbdf904a5c2d7e86f63da217c5b25b951baf0a  guix-build-a5180bffbe78/output/arm-linux-gnueabihf/SHA256SUMS.part
1af95de6c98dbb2ad9918be89814faa12929f14c5ce9ecd6b53e094611878436  guix-build-a5180bffbe78/output/arm-linux-gnueabihf/bitcoin-a5180bffbe78-arm-linux-gnueabihf-debug.tar.gz
3071eb8bfd6be7ce14ddf9f564aac756530c271b10027d16b1b34beb6f1929f1  guix-build-a5180bffbe78/output/arm-linux-gnueabihf/bitcoin-a5180bffbe78-arm-linux-gnueabihf.tar.gz
44ea08071ffe9cd5d3b222f8ecd864a39e3d1ad4d8f86c46b286f070c4197d43  guix-build-a5180bffbe78/output/arm64-apple-darwin/SHA256SUMS.part
cdc360e2a3d79acca40611dd8b684decbecb7a1f3031e7d41daafd7467430ddc  guix-build-a5180bffbe78/output/arm64-apple-darwin/bitcoin-a5180bffbe78-arm64-apple-darwin-unsigned.tar.gz
23d2af4192ad1b09a15db9b9df297f14679222a21ff6e87fefc7bf2c30a5003f  guix-build-a5180bffbe78/output/arm64-apple-darwin/bitcoin-a5180bffbe78-arm64-apple-darwin-unsigned.zip
c13a6bd4e7e84894b6f192dd93dba6921b20230417aab482ce0d48de2974e66e  guix-build-a5180bffbe78/output/arm64-apple-darwin/bitcoin-a5180bffbe78-arm64-apple-darwin.tar.gz
6790b7c1ae5225a41d01711e8e9c6c4540b4630602e1c03e93398f7259d2a918  guix-build-a5180bffbe78/output/dist-archive/bitcoin-a5180bffbe78.tar.gz
c43ab5fba3b9821eb47a677898b1fb9570738f0dff44a1b8810213c3a1701f87  guix-build-a5180bffbe78/output/powerpc64-linux-gnu/SHA256SUMS.part
43a4a7f7b6de83b48f2ee50cc9d9d69773e3a25d82df6ea0d2a29f6f9a9d6a29  guix-build-a5180bffbe78/output/powerpc64-linux-gnu/bitcoin-a5180bffbe78-powerpc64-linux-gnu-debug.tar.gz
bd2fa8392e19dac22b73cbcef8075fe7ad5ff6de254f4b98cd9b36b27fd213b8  guix-build-a5180bffbe78/output/powerpc64-linux-gnu/bitcoin-a5180bffbe78-powerpc64-linux-gnu.tar.gz
731d0c5d31bc6effd8d9b947cdf403f3da52c1a7782bc1b64050548efad210b9  guix-build-a5180bffbe78/output/riscv64-linux-gnu/SHA256SUMS.part
e8c70d9b0738094a902e6af812f2ceca4b8150e8f015b80e73e12d67b9302b44  guix-build-a5180bffbe78/output/riscv64-linux-gnu/bitcoin-a5180bffbe78-riscv64-linux-gnu-debug.tar.gz
2fc1b988921bb3af816042e53d6f9d2ab24a71d63f4b287f270d900da147a87a  guix-build-a5180bffbe78/output/riscv64-linux-gnu/bitcoin-a5180bffbe78-riscv64-linux-gnu.tar.gz
932ce059509afa607ca44113147748a3b7d6c527d7ea618e4f8702bbae619132  guix-build-a5180bffbe78/output/x86_64-apple-darwin/SHA256SUMS.part
d7169b5ed1edb3d2958955e8dc3022a05a62fc96fd4cfff6bb5d7cc2ea5d3410  guix-build-a5180bffbe78/output/x86_64-apple-darwin/bitcoin-a5180bffbe78-x86_64-apple-darwin-unsigned.tar.gz
a44164c8f35ad3cbf9db98ecdebed7da63635d0c64898473fe289656ae852178  guix-build-a5180bffbe78/output/x86_64-apple-darwin/bitcoin-a5180bffbe78-x86_64-apple-darwin-unsigned.zip
34f8b2d2f30154bc5495ee312cc6ca7f891e489e4019c1daa75b04ba99f1be9c  guix-build-a5180bffbe78/output/x86_64-apple-darwin/bitcoin-a5180bffbe78-x86_64-apple-darwin.tar.gz
9f050eb2e524be24c2dc0364e1c6e686c1b2695efbba83cdd205aff75fff3494  guix-build-a5180bffbe78/output/x86_64-linux-gnu/SHA256SUMS.part
29e543875d1c78dea2cbb25ef61a2dd92ab67e80b8d36ab0d8ff70bb0d9041a5  guix-build-a5180bffbe78/output/x86_64-linux-gnu/bitcoin-a5180bffbe78-x86_64-linux-gnu-debug.tar.gz
da2c1fc38fa75b03a17f212b5cb320f6e319bc46ec247baccc8caf59d6cee5ce  guix-build-a5180bffbe78/output/x86_64-linux-gnu/bitcoin-a5180bffbe78-x86_64-linux-gnu.tar.gz
5ca7b25cd4f1ef71567a818be7ad4937bf47d635678fc372af6df34e2fbafba8  guix-build-a5180bffbe78/output/x86_64-w64-mingw32/SHA256SUMS.part
6282c55ed5f8d059a866ba8fb05970e01d5531786e251f2da7cfb4f5f851d932  guix-build-a5180bffbe78/output/x86_64-w64-mingw32/bitcoin-a5180bffbe78-win64-debug.zip
dc53040e8e1342418e55361c051686d0637cdaa6cb6e6cd1f40b99d9f9045975  guix-build-a5180bffbe78/output/x86_64-w64-mingw32/bitcoin-a5180bffbe78-win64-setup-unsigned.exe
00d4f7c42dd21fc3900bdd1e996bf74b2a2b8ddd405674b062ca9e5ea067cf7d  guix-build-a5180bffbe78/output/x86_64-w64-mingw32/bitcoin-a5180bffbe78-win64-unsigned.tar.gz
58bd22e8ff44570ae0f838560544c17f5aa2e3792bd01bbdba5683ca517bc7ed  guix-build-a5180bffbe78/output/x86_64-w64-mingw32/bitcoin-a5180bffbe78-win64.zip
```

File: depends/packages/libevent.mk [hunk 1]
```diff
@@ -3,27 +3,30 @@ $(package)_version=2.1.12-stable
...
 # is also required to enable support for ipv6. See #19375.
 define $(package)_set_vars
-  $(package)_config_opts=--disable-shared --disable-openssl --disable-libevent-regress --disable-samples
-  $(package)_config_opts += --disable-dependency-tracking --enable-option-checking
-  $(package)_config_opts_release=--disable-debug-mode
+  $(package)_config_opts=-DEVENT__DISABLE_BENCHMARK=ON -DEVENT__DISABLE_OPENSSL=ON
+  $(package)_config_opts+=-DEVENT__DISABLE_SAMPLES=ON -DEVENT__DISABLE_REGRESS=ON
+  $(package)_config_opts+=-DEVENT__DISABLE_TESTS=ON -DEVENT__LIBRARY_TYPE=STATIC
   $(package)_cppflags_mingw32=-D_WIN32_WINNT=0x0601
+  $(package)_config_opts_debug+=-DEVENT__DISABLE_DEBUG_MODE=ON
```
Comment (TheCharlatan) [inline depends/packages/libevent.mk:17]: Could this just always be disabled?

Comment (fanquake) [reply depends/packages/libevent.mk:17]: Probably. Happy to just drop this, and it could always be added later.

Comment (TheCharlatan) [reply depends/packages/libevent.mk]: That's fine too, it is not used anyway.

**Comment by TheCharlatan** [comment]
Guix builds (aarch64):
```
d702d02df48bc540da55c47ca7110d122a27ba179ab728fb8bdb6e27589f754c  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/SHA256SUMS.part
d806e1994bc873a975714ce75bca87fd3fcec4055e24d1ed0afe99ebc3503288  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/bitcoin-f59e9057e2aa-aarch64-linux-gnu-debug.tar.gz
a9844e6a4d3e86df69b98281c942ac01136b4b115fc6a2504740368c5d441fbd  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/bitcoin-f59e9057e2aa-aarch64-linux-gnu.tar.gz
4e3ea3b82c4c46c1539ba6cfe6627e331e50feda973d0627598671c03fd727f4  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/SHA256SUMS.part
99712894848ef7a0b90b3a8582d617809e986abda202e9d4941ee970ca0d9a3d  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/bitcoin-f59e9057e2aa-arm-linux-gnueabihf-debug.tar.gz
aec34ff47cf315949f2ddbf4eb0ed69dd2ec278b20ab7909a16ce48abd41e7ee  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/bitcoin-f59e9057e2aa-arm-linux-gnueabihf.tar.gz
072b0091fb89e726931fe214b7b2f03e188f64c67df86ae1342354713697eea8  guix-build-f59e9057e2aa/output/arm64-apple-darwin/SHA256SUMS.part
8cdcab079ab725e3d823592c0f1b11a900515bc70c949605f5750e9666cccd02  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.tar.gz
85d962a1a068ce77bb689af1a7ddcbdb347928f7e80dc7fb461d6d49d278e72a  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.zip
7693aa014666780dad65c51e25bca5f2c67682475b40c570e30515f75ec73fdb  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin.tar.gz
081d771bff0deefa9a298df40b8e564a77d527cf6be3268402ee3746b19156a2  guix-build-f59e9057e2aa/output/dist-archive/bitcoin-f59e9057e2aa.tar.gz
6bcae84c9b5e1a74e9ad490af8a020b0baf567b7e1ac798af1f37755de699169  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/SHA256SUMS.part
aac0fd663a9418a016ded828a3961371f17b8cec19306ef278061c02b15308fc  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/bitcoin-f59e9057e2aa-powerpc64-linux-gnu-debug.tar.gz
56b9c6e5b114701dc3f862414343a0c41c840795706f9289e17fd4967bd72a7b  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/bitcoin-f59e9057e2aa-powerpc64-linux-gnu.tar.gz
6b0eb35a6af793049302ab284999c143d23b9f60080070c221ae93fd46ee4715  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/SHA256SUMS.part
b6a6948fea45d1fb571029264f3d406b0c1846d3e47bd2929e12096372f33385  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/bitcoin-f59e9057e2aa-riscv64-linux-gnu-debug.tar.gz
125670ea7e908757ce988faf6f9c815d9e74ccbc9bf67f2f1eee834ddc97f602  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/bitcoin-f59e9057e2aa-riscv64-linux-gnu.tar.gz
59e2f661e5fcc44eaed1476618f765225609a1f2d5776c4b9f88bd7fdfb36fb6  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/SHA256SUMS.part
e7486f0a29beb0c14f3949b7b2dc6387842d94d3c7379db8e8bd1c7cde221a07  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.tar.gz
a5bc7e46e4663a4de58e4ddc22b072a0987cfa4a5996b67f059b06b573bb0e48  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.zip
71a1359a0d51029c3757e908c5580544e5b00d47bdff3ebcfbed6f3f38f68fe5  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin.tar.gz
56b9393b3002e8ee9aee46e09b0e8a2de8aeb99a1c7d5d3271e1b7ef19d01282  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/SHA256SUMS.part
67cb5897c74067400a788f3d7f5ee72c76c2500c8f690be1c30a673fbb4dc083  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/bitcoin-f59e9057e2aa-x86_64-linux-gnu-debug.tar.gz
ac384de91a8c93a661a639817e86aeb431e881e86582ff1c22fe1da2e1c353bf  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/bitcoin-f59e9057e2aa-x86_64-linux-gnu.tar.gz
3fd25c0a6270115f4680aa97009497f14d7912736e58715875ab0b750a84430b  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/SHA256SUMS.part
e70a51f884fc57420b5d4bcbf72d1b2ea84368015cc6634374d6059f5fc31eab  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-debug.zip
988e4e0389834169b1b372d65a6daec712ba6fcb48c1ecd7b5694103c92bbd9e  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-setup-unsigned.exe
9952ba40daaf6e123fffe3bd45e563c57d2a50fc7cbadd9771b9dc8befc514b0  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-unsigned.tar.gz
64a7c44fbd4941ba991368fcf4c69151e8f0f6fdd4bb563312e3ea47b4fd2f37  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64.zip
```

**Review by hebasto — COMMENTED** [review]
My Guix build:
```
x86_64
d702d02df48bc540da55c47ca7110d122a27ba179ab728fb8bdb6e27589f754c  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/SHA256SUMS.part
d806e1994bc873a975714ce75bca87fd3fcec4055e24d1ed0afe99ebc3503288  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/bitcoin-f59e9057e2aa-aarch64-linux-gnu-debug.tar.gz
a9844e6a4d3e86df69b98281c942ac01136b4b115fc6a2504740368c5d441fbd  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/bitcoin-f59e9057e2aa-aarch64-linux-gnu.tar.gz
4e3ea3b82c4c46c1539ba6cfe6627e331e50feda973d0627598671c03fd727f4  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/SHA256SUMS.part
99712894848ef7a0b90b3a8582d617809e986abda202e9d4941ee970ca0d9a3d  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/bitcoin-f59e9057e2aa-arm-linux-gnueabihf-debug.tar.gz
aec34ff47cf315949f2ddbf4eb0ed69dd2ec278b20ab7909a16ce48abd41e7ee  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/bitcoin-f59e9057e2aa-arm-linux-gnueabihf.tar.gz
072b0091fb89e726931fe214b7b2f03e188f64c67df86ae1342354713697eea8  guix-build-f59e9057e2aa/output/arm64-apple-darwin/SHA256SUMS.part
8cdcab079ab725e3d823592c0f1b11a900515bc70c949605f5750e9666cccd02  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.tar.gz
85d962a1a068ce77bb689af1a7ddcbdb347928f7e80dc7fb461d6d49d278e72a  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.zip
7693aa014666780dad65c51e25bca5f2c67682475b40c570e30515f75ec73fdb  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin.tar.gz
081d771bff0deefa9a298df40b8e564a77d527cf6be3268402ee3746b19156a2  guix-build-f59e9057e2aa/output/dist-archive/bitcoin-f59e9057e2aa.tar.gz
6bcae84c9b5e1a74e9ad490af8a020b0baf567b7e1ac798af1f37755de699169  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/SHA256SUMS.part
aac0fd663a9418a016ded828a3961371f17b8cec19306ef278061c02b15308fc  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/bitcoin-f59e9057e2aa-powerpc64-linux-gnu-debug.tar.gz
56b9c6e5b114701dc3f862414343a0c41c840795706f9289e17fd4967bd72a7b  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/bitcoin-f59e9057e2aa-powerpc64-linux-gnu.tar.gz
6b0eb35a6af793049302ab284999c143d23b9f60080070c221ae93fd46ee4715  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/SHA256SUMS.part
b6a6948fea45d1fb571029264f3d406b0c1846d3e47bd2929e12096372f33385  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/bitcoin-f59e9057e2aa-riscv64-linux-gnu-debug.tar.gz
125670ea7e908757ce988faf6f9c815d9e74ccbc9bf67f2f1eee834ddc97f602  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/bitcoin-f59e9057e2aa-riscv64-linux-gnu.tar.gz
59e2f661e5fcc44eaed1476618f765225609a1f2d5776c4b9f88bd7fdfb36fb6  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/SHA256SUMS.part
e7486f0a29beb0c14f3949b7b2dc6387842d94d3c7379db8e8bd1c7cde221a07  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.tar.gz
a5bc7e46e4663a4de58e4ddc22b072a0987cfa4a5996b67f059b06b573bb0e48  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.zip
71a1359a0d51029c3757e908c5580544e5b00d47bdff3ebcfbed6f3f38f68fe5  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin.tar.gz
56b9393b3002e8ee9aee46e09b0e8a2de8aeb99a1c7d5d3271e1b7ef19d01282  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/SHA256SUMS.part
67cb5897c74067400a788f3d7f5ee72c76c2500c8f690be1c30a673fbb4dc083  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/bitcoin-f59e9057e2aa-x86_64-linux-gnu-debug.tar.gz
ac384de91a8c93a661a639817e86aeb431e881e86582ff1c22fe1da2e1c353bf  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/bitcoin-f59e9057e2aa-x86_64-linux-gnu.tar.gz
3fd25c0a6270115f4680aa97009497f14d7912736e58715875ab0b750a84430b  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/SHA256SUMS.part
e70a51f884fc57420b5d4bcbf72d1b2ea84368015cc6634374d6059f5fc31eab  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-debug.zip
988e4e0389834169b1b372d65a6daec712ba6fcb48c1ecd7b5694103c92bbd9e  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-setup-unsigned.exe
9952ba40daaf6e123fffe3bd45e563c57d2a50fc7cbadd9771b9dc8befc514b0  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-unsigned.tar.gz
64a7c44fbd4941ba991368fcf4c69151e8f0f6fdd4bb563312e3ea47b4fd2f37  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64.zip
```

**Review by sedited — APPROVED** [review]
ACK f59e9057e2aa596b54cf9e85bab35c3ead137547

**Comment by willcl-ark** [comment]
My build on x86_64 (excluding SHA256SUMS.part):

```
d806e1994bc873a975714ce75bca87fd3fcec4055e24d1ed0afe99ebc3503288  bitcoin-f59e9057e2aa-aarch64-linux-gnu-debug.tar.gz
a9844e6a4d3e86df69b98281c942ac01136b4b115fc6a2504740368c5d441fbd  bitcoin-f59e9057e2aa-aarch64-linux-gnu.tar.gz
99712894848ef7a0b90b3a8582d617809e986abda202e9d4941ee970ca0d9a3d  bitcoin-f59e9057e2aa-arm-linux-gnueabihf-debug.tar.gz
aec34ff47cf315949f2ddbf4eb0ed69dd2ec278b20ab7909a16ce48abd41e7ee  bitcoin-f59e9057e2aa-arm-linux-gnueabihf.tar.gz
8cdcab079ab725e3d823592c0f1b11a900515bc70c949605f5750e9666cccd02  bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.tar.gz
85d962a1a068ce77bb689af1a7ddcbdb347928f7e80dc7fb461d6d49d278e72a  bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.zip
7693aa014666780dad65c51e25bca5f2c67682475b40c570e30515f75ec73fdb  bitcoin-f59e9057e2aa-arm64-apple-darwin.tar.gz
081d771bff0deefa9a298df40b8e564a77d527cf6be3268402ee3746b19156a2  bitcoin-f59e9057e2aa.tar.gz
aac0fd663a9418a016ded828a3961371f17b8cec19306ef278061c02b15308fc  bitcoin-f59e9057e2aa-powerpc64-linux-gnu-debug.tar.gz
56b9c6e5b114701dc3f862414343a0c41c840795706f9289e17fd4967bd72a7b  bitcoin-f59e9057e2aa-powerpc64-linux-gnu.tar.gz
b6a6948fea45d1fb571029264f3d406b0c1846d3e47bd2929e12096372f33385  bitcoin-f59e9057e2aa-riscv64-linux-gnu-debug.tar.gz
125670ea7e908757ce988faf6f9c815d9e74ccbc9bf67f2f1eee834ddc97f602  bitcoin-f59e9057e2aa-riscv64-linux-gnu.tar.gz
e7486f0a29beb0c14f3949b7b2dc6387842d94d3c7379db8e8bd1c7cde221a07  bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.tar.gz
a5bc7e46e4663a4de58e4ddc22b072a0987cfa4a5996b67f059b06b573bb0e48  bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.zip
71a1359a0d51029c3757e908c5580544e5b00d47bdff3ebcfbed6f3f38f68fe5  bitcoin-f59e9057e2aa-x86_64-apple-darwin.tar.gz
67cb5897c74067400a788f3d7f5ee72c76c2500c8f690be1c30a673fbb4dc083  bitcoin-f59e9057e2aa-x86_64-linux-gnu-debug.tar.gz
ac384de91a8c93a661a639817e86aeb431e881e86582ff1c22fe1da2e1c353bf  bitcoin-f59e9057e2aa-x86_64-linux-gnu.tar.gz
e70a51f884fc57420b5d4bcbf72d1b2ea84368015cc6634374d6059f5fc31eab  bitcoin-f59e9057e2aa-win64-debug.zip
988e4e0389834169b1b372d65a6daec712ba6fcb48c1ecd7b5694103c92bbd9e  bitcoin-f59e9057e2aa-win64-setup-unsigned.exe
9952ba40daaf6e123fffe3bd45e563c57d2a50fc7cbadd9771b9dc8befc514b0  bitcoin-f59e9057e2aa-win64-unsigned.tar.gz
64a7c44fbd4941ba991368fcf4c69151e8f0f6fdd4bb563312e3ea47b4fd2f37  bitcoin-f59e9057e2aa-win64.zip
```

**Review by willcl-ark — APPROVED** [review]
ACK f59e9057e2aa596b54cf9e85bab35c3ead137547

Tested locally and with guix build. Changes all look good to me, with equivalent options being set where possible.

There are not direct equivalents of `--disable-dependency-tracking` and `--enable-option-checking` for cmake, but this doesn't seem like a problem for us here.

**Comment by m3dwards** [comment]
Post merge Guix build matched:

```shell
d702d02df48bc540da55c47ca7110d122a27ba179ab728fb8bdb6e27589f754c  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/SHA256SUMS.part
d806e1994bc873a975714ce75bca87fd3fcec4055e24d1ed0afe99ebc3503288  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/bitcoin-f59e9057e2aa-aarch64-linux-gnu-debug.tar.gz
a9844e6a4d3e86df69b98281c942ac01136b4b115fc6a2504740368c5d441fbd  guix-build-f59e9057e2aa/output/aarch64-linux-gnu/bitcoin-f59e9057e2aa-aarch64-linux-gnu.tar.gz
4e3ea3b82c4c46c1539ba6cfe6627e331e50feda973d0627598671c03fd727f4  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/SHA256SUMS.part
99712894848ef7a0b90b3a8582d617809e986abda202e9d4941ee970ca0d9a3d  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/bitcoin-f59e9057e2aa-arm-linux-gnueabihf-debug.tar.gz
aec34ff47cf315949f2ddbf4eb0ed69dd2ec278b20ab7909a16ce48abd41e7ee  guix-build-f59e9057e2aa/output/arm-linux-gnueabihf/bitcoin-f59e9057e2aa-arm-linux-gnueabihf.tar.gz
072b0091fb89e726931fe214b7b2f03e188f64c67df86ae1342354713697eea8  guix-build-f59e9057e2aa/output/arm64-apple-darwin/SHA256SUMS.part
8cdcab079ab725e3d823592c0f1b11a900515bc70c949605f5750e9666cccd02  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.tar.gz
85d962a1a068ce77bb689af1a7ddcbdb347928f7e80dc7fb461d6d49d278e72a  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin-unsigned.zip
7693aa014666780dad65c51e25bca5f2c67682475b40c570e30515f75ec73fdb  guix-build-f59e9057e2aa/output/arm64-apple-darwin/bitcoin-f59e9057e2aa-arm64-apple-darwin.tar.gz
081d771bff0deefa9a298df40b8e564a77d527cf6be3268402ee3746b19156a2  guix-build-f59e9057e2aa/output/dist-archive/bitcoin-f59e9057e2aa.tar.gz
6bcae84c9b5e1a74e9ad490af8a020b0baf567b7e1ac798af1f37755de699169  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/SHA256SUMS.part
aac0fd663a9418a016ded828a3961371f17b8cec19306ef278061c02b15308fc  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/bitcoin-f59e9057e2aa-powerpc64-linux-gnu-debug.tar.gz
56b9c6e5b114701dc3f862414343a0c41c840795706f9289e17fd4967bd72a7b  guix-build-f59e9057e2aa/output/powerpc64-linux-gnu/bitcoin-f59e9057e2aa-powerpc64-linux-gnu.tar.gz
6b0eb35a6af793049302ab284999c143d23b9f60080070c221ae93fd46ee4715  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/SHA256SUMS.part
b6a6948fea45d1fb571029264f3d406b0c1846d3e47bd2929e12096372f33385  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/bitcoin-f59e9057e2aa-riscv64-linux-gnu-debug.tar.gz
125670ea7e908757ce988faf6f9c815d9e74ccbc9bf67f2f1eee834ddc97f602  guix-build-f59e9057e2aa/output/riscv64-linux-gnu/bitcoin-f59e9057e2aa-riscv64-linux-gnu.tar.gz
59e2f661e5fcc44eaed1476618f765225609a1f2d5776c4b9f88bd7fdfb36fb6  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/SHA256SUMS.part
e7486f0a29beb0c14f3949b7b2dc6387842d94d3c7379db8e8bd1c7cde221a07  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.tar.gz
a5bc7e46e4663a4de58e4ddc22b072a0987cfa4a5996b67f059b06b573bb0e48  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin-unsigned.zip
71a1359a0d51029c3757e908c5580544e5b00d47bdff3ebcfbed6f3f38f68fe5  guix-build-f59e9057e2aa/output/x86_64-apple-darwin/bitcoin-f59e9057e2aa-x86_64-apple-darwin.tar.gz
56b9393b3002e8ee9aee46e09b0e8a2de8aeb99a1c7d5d3271e1b7ef19d01282  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/SHA256SUMS.part
67cb5897c74067400a788f3d7f5ee72c76c2500c8f690be1c30a673fbb4dc083  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/bitcoin-f59e9057e2aa-x86_64-linux-gnu-debug.tar.gz
ac384de91a8c93a661a639817e86aeb431e881e86582ff1c22fe1da2e1c353bf  guix-build-f59e9057e2aa/output/x86_64-linux-gnu/bitcoin-f59e9057e2aa-x86_64-linux-gnu.tar.gz
3fd25c0a6270115f4680aa97009497f14d7912736e58715875ab0b750a84430b  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/SHA256SUMS.part
e70a51f884fc57420b5d4bcbf72d1b2ea84368015cc6634374d6059f5fc31eab  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-debug.zip
988e4e0389834169b1b372d65a6daec712ba6fcb48c1ecd7b5694103c92bbd9e  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-setup-unsigned.exe
9952ba40daaf6e123fffe3bd45e563c57d2a50fc7cbadd9771b9dc8befc514b0  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64-unsigned.tar.gz
64a7c44fbd4941ba991368fcf4c69151e8f0f6fdd4bb563312e3ea47b4fd2f37  guix-build-f59e9057e2aa/output/x86_64-w64-mingw32/bitcoin-f59e9057e2aa-win64.zip
```

**Comment by hebasto** [comment]
Ported to the CMake-based build system in https://github.com/hebasto/bitcoin/pull/262.

**Pull request merged**